src/vma/dev/ring_eth_cb.cpp

```diff
@@ -121,6 +123,11 @@ void ring_eth_cb::create_resources(ring_resource_creation_info_t *p_ring_info,
 			m_single_wqe_log_num_of_strides = mp_rq_caps->min_single_wqe_log_num_of_strides;
 		}
 	}
+	// FW bug only exists in CX5 if more then 2048 (1<<11) strides in WQE
+	if (m_single_wqe_log_num_of_strides < 12 ||
+	    r_ibv_dev_attr.vendor_part_id == MLX_CX4_DEV_ID) {
+		first_call = false;
```

Comment:
@rafiw as I know there is no direct link between FW version and DeviceID. So the same device can have different FW versions. Your current check is based on assumption that all devices with this DeviceID have the issue. Probably you need to check `fw_ver` ot `hw_ver`.